### DISCUSSION TITLE
how can we make it easier to read and interpret the results?

### DISCUSSION THREAD

**Discussion by BurntSushi** [BurntSushi/rebar#4] [Q&A]
Even for myself, I find it difficult to digest the actual numeric results in the tables in the README. My hope was that the analysis prose would help offset this, but it's still useful to be able to read the raw numbers.

**Comment by BurntSushi** [comment]
@matthieu-m on reddit [said](https://old.reddit.com/r/rust/comments/14rcng5/burntsushirebar_a_biased_barometer_for_gauging/jqs2ows/):

I find reading the summary arrays for each section pretty difficult, between the engines not being ordered by "performance" and the changing units, which make it really hard to see what the groups (Good/Average/Bad) are.

I was thinking that maybe it would be possible to identify such groups with background colors/icons for each cell. For example:

 - Tick mark + green background for the "best in class" in a particular column.
 - Cross mark + red background for the "worst in class" in a particular column.

And the average ones get nothing, as they're not particularly noteworthy.

(Alternatively, a cup icon for the winners, and a skull for the losers...)

As for best-vs-worse, I think it would be most useful to use thresholds, rather than a specific count. For example, any engine within 10x of the best is in the same order of magnitude, so gets the best-in-class treatment, and conversely, any engine below 100x or 1000x of the best is for away that they get the worst-in-class treatment.

**Reply by bstrie** [reply]
I would suggest using a bar graph to represent the data in each table, although alas Mermaid doesn't support bar graphs so you'd have to generate them manually.

**Reply by epompeii** [reply]
Line graphs are possible with Bencher (https://github.com/BurntSushi/rebar/pull/2), though this likely needs some updating.

**Reply by bstrie** [reply]
I.e. something like: 
![bar-graph](https://github.com/BurntSushi/rebar/assets/865233/517f9621-65cc-4d52-a4f2-67d01ce7384f)

**Reply by epompeii** [reply]
Definitely! This is on the roadmap: https://github.com/bencherdev/bencher/issues/72
